### PR TITLE
allow Configuration to customize the transport of checkin and payment…

### PR DESCRIPTION
**Pull request by IhmeHippi**
…order requests

### CODE REVIEW

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-android/pullRequest?prid=6212608&bid=20258447)

File: sdk/src/main/java/com/swedbankpay/mobilesdk/internal/InternalPaymentViewModel.kt [hunk 1]
```diff
@@ -258,24 +264,66 @@ internal class InternalPaymentViewModel(app: Application) : AndroidViewModel(app
...
-        class HtmlContent(@StringRes private val template: Int, private val link: String) : UIState() {
-            override fun getWebViewPage(vm: AndroidViewModel): String? {
-                return vm.getApplication<Application>().getString(template, link)
+        class HtmlContent(
+            val baseUrl: String?,
+            @StringRes private val template: Int,
+            private val link: String
+        ) : UIState() {
+            override val asHtmlContent get() = this
+            fun getWebViewPage(context: Context): String {
```
Comment (payex-dev) [inline sdk/src/main/java/com/swedbankpay/mobilesdk/internal/InternalPaymentViewModel.kt:279]: Codacy found an issue: [UndocumentedPublicFunction - [getWebViewPage] at /src/sdk/src/main/java/com/swedbankpay/mobilesdk/internal/InternalPaymentViewModel.kt:279:17](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-android/pullRequest?prid=6212608)

File: sdk/src/main/java/com/swedbankpay/mobilesdk/RequestDecorator.kt [hunk 2]
```diff
@@ -30,6 +34,49 @@ package com.swedbankpay.mobilesdk
...
+        @JvmStatic
+        fun withHeaders(headers: Map<String, String>): RequestDecorator {
+            val namesAndValues = arrayOfNulls<String>(2 * headers.size)
+            var i = 0
+            for ((name, value) in headers) {
+                namesAndValues[i++] = name
+                namesAndValues[i++] = value
+            }
+            @Suppress("UNCHECKED_CAST")
+            return withHeaders(*namesAndValues as Array<String>)
```
Comment (payex-dev) [inline sdk/src/main/java/com/swedbankpay/mobilesdk/RequestDecorator.kt:77]: Codacy found an issue: [SpreadOperator - [withHeaders] at /src/sdk/src/main/java/com/swedbankpay/mobilesdk/RequestDecorator.kt:77:31](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-android/pullRequest?prid=6212608)

File: sdk/src/main/java/com/swedbankpay/mobilesdk/internal/InternalPaymentViewModel.kt (hunk 1 above)
Comment (IhmeHippi) [reply sdk/src/main/java/com/swedbankpay/mobilesdk/internal/InternalPaymentViewModel.kt:279]: This is an internal function, not a public one

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-android/pullRequest?prid=6212608&bid=20258447)

File: sdk/src/main/java/com/swedbankpay/mobilesdk/internal/InternalPaymentViewModel.kt (hunk 1 above)
Comment (payex-dev) [inline sdk/src/main/java/com/swedbankpay/mobilesdk/internal/InternalPaymentViewModel.kt:279]: Codacy found an issue: [UndocumentedPublicFunction - [getWebViewPage] at /src/sdk/src/main/java/com/swedbankpay/mobilesdk/internal/InternalPaymentViewModel.kt:279:17](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-android/pullRequest?prid=6212608)

**Pull request merged**